src/nbl/asset/utils/ISPIRVDebloater.cpp

```diff
@@ -8,23 +8,92 @@
 
 using namespace nbl::asset;
 
+static constexpr spv_target_env SPIRV_VERSION = spv_target_env::SPV_ENV_UNIVERSAL_1_6;
+
 ISPIRVDebloater::ISPIRVDebloater()
 {
     constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
         ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
```

Comment:
why are you doing two rounds of dead branch elimination followed by dead function elimination ?

Reply:
leave a comment in the code if you still think that is corect